### PR TITLE
Implement object storage credential

### PR DESCRIPTION
**Pull request by hcsa73**

### CODE REVIEW

File: stackit/internal/services/objectstorage/objectstorage_acc_test.go [hunk 1]
```diff
@@ -107,19 +140,34 @@ func TestAccObjectStorageResource(t *testing.T) {
 						"data.stackit_objectstorage_bucket.bucket", "url_virtual_hosted_style",
 					),
 
-					// Credentials group data
-					resource.TestCheckResourceAttr("data.stackit_objectstorage_credentials_group.credentials_group", "project_id", credentialsGroupResource["project_id"]),
+					// Credential group data
```
Comment (joaopalet) [inline stackit/internal/services/objectstorage/objectstorage_acc_test.go:143]: I think here it's fine to leave it plural (as it os on line 83) as it's the "credentials group"

File: stackit/internal/services/objectstorage/credential/resource.go [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &credentialResource{}
+	_ resource.ResourceWithConfigure   = &credentialResource{}
+	_ resource.ResourceWithImportState = &credentialResource{}
+)
+
+type Model struct {
+	Id                  types.String      `tfsdk:"id"` // needed by TF
+	CredentialId        types.String      `tfsdk:"credentials_id"`
```
Comment (joaopalet) [inline stackit/internal/services/objectstorage/credential/resource.go:34]:
```suggestion
	CredentialId        types.String      `tfsdk:"credential_id"`
```

File: stackit/internal/services/objectstorage/credential/resource.go [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+	_, err := r.client.DeleteAccessKey(ctx, projectId, credentialId).CredentialsGroup(credentialsGroupId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting credential", fmt.Sprintf("Calling API: %v", err))
+	}
+
+	tflog.Info(ctx, "ObjectStorage credential deleted")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: project_id,instance_id,credentials_id
```
Comment (joaopalet) [inline stackit/internal/services/objectstorage/credential/resource.go:286]:
```suggestion
// The expected format of the resource import identifier is: project_id,credentials_group_id,credential_id
```

**Review by joaopalet — CHANGES_REQUESTED** [review]

File: stackit/internal/services/objectstorage/credential/resource.go [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+}
+
+// Schema defines the schema for the resource.
+func (r *credentialResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":                 "ObjectStorage credential resource schema.",
+		"id":                   "Terraform's internal resource identifier. It is structured as \"`project_id`,`credentials_group_id`,`credential_id`\".",
+		"credential_id":        "The credential ID.",
+		"credentials_group_id": "The credential group ID.",
+		"project_id":           "STACKIT Project ID to which the credential group is associated.",
```
[the remark this answers is not in the document]
Comment (hcsa73) [reply stackit/internal/services/objectstorage/credential/resource.go:101]: Technically the credentials group is linked to a project, and the credential is linked to a credentials group

File: stackit/internal/services/objectstorage/credential/datasource.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+// Schema defines the schema for the datasource.
+func (r *credentialDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":                 "ObjectStorage credential data source schema.",
+		"id":                   "Terraform's internal resource identifier. It is structured as \"`project_id`,`credentials_group_id`,`credential_id`\".",
+		"credential_id":        "The credential ID.",
+		"credentials_group_id": "The credential group ID.",
+		"project_id":           "STACKIT Project ID to which the credential group is associated.",
```
[the remark this answers is not in the document]
Comment (hcsa73) [reply stackit/internal/services/objectstorage/credential/datasource.go:80]: Technically the credentials group is linked to a project, and the credential is linked to a credentials group

**Review by joaopalet — APPROVED** [review]

**Pull request merged**